### ISSUE TITLE
VerifyError on Azul Zing JDK when mocking enum with overriden method

### ISSUE THREAD

**Issue by aswath80** [raphw/byte-buddy#1502]
The following sample passes on Zulu JDK but fails with Zing (Azul Prime) that is supported by mockito-inline as per the documentation.

JDK: zingjdk_17-23_02_200
Mockito: 4.10.0
Byte-buddy: 1.12.19 & 1.14.5

```
import static org.mockito.Mockito.spy;

public class Test {

    static enum TestEnum {

        X(1) {
            @Override
            void print() {
                System.out.println("Override: " + this.z);
            }
        },
        Y(2);

        final int z;

        TestEnum(int z) {
            this.z = z;
        }

        void print() {
            System.out.println("Original: " + this.z);
        }
    }

    public static void main(String[] args) {
        TestEnum testEnum = spy(TestEnum.X);
        System.out.println(testEnum);
    }
}
```

Error:

```
Exception in thread "main" org.mockito.exceptions.base.MockitoException:
Mockito cannot mock this class: class Test$TestEnum$1.
Can not mock final classes with the following settings :
 - explicit serialization (e.g. withSettings().serializable())
 - extra interfaces (e.g. withSettings().extraInterfaces(...))

You are seeing this disclaimer because Mockito is configured to create inlined mocks.
You can learn about inline mocks and their limitations under item #39 of the Mockito class javadoc.

Underlying exception : java.lang.IllegalArgumentException: Could not create type
	at Test.main(Test.java:27)
Caused by: java.lang.IllegalArgumentException: Could not create type
	at net.bytebuddy.TypeCache.findOrInsert(TypeCache.java:170)
	at net.bytebuddy.TypeCache$WithInlineExpunction.findOrInsert(TypeCache.java:399)
	at net.bytebuddy.TypeCache.findOrInsert(TypeCache.java:190)
	at net.bytebuddy.TypeCache$WithInlineExpunction.findOrInsert(TypeCache.java:410)
	at org.mockito.internal.creation.bytebuddy.TypeCachingBytecodeGenerator.mockClass(TypeCachingBytecodeGenerator.java:40)
	at org.mockito.internal.creation.bytebuddy.InlineDelegateByteBuddyMockMaker.createMockType(InlineDelegateByteBuddyMockMaker.java:396)
	at org.mockito.internal.creation.bytebuddy.InlineDelegateByteBuddyMockMaker.doCreateMock(InlineDelegateByteBuddyMockMaker.java:355)
	at org.mockito.internal.creation.bytebuddy.InlineDelegateByteBuddyMockMaker.createSpy(InlineDelegateByteBuddyMockMaker.java:345)
	at org.mockito.internal.creation.bytebuddy.InlineByteBuddyMockMaker.createSpy(InlineByteBuddyMockMaker.java:62)
	at org.mockito.internal.util.MockUtil.createMock(MockUtil.java:91)
	at org.mockito.internal.MockitoCore.mock(MockitoCore.java:88)
	at org.mockito.Mockito.spy(Mockito.java:2120)
	... 1 more
Caused by: java.lang.VerifyError
	at java.instrument/sun.instrument.InstrumentationImpl.retransformClasses0(Native Method)
	at java.instrument/sun.instrument.InstrumentationImpl.retransformClasses(InstrumentationImpl.java:169)
	at org.mockito.internal.creation.bytebuddy.InlineBytecodeGenerator.triggerRetransformation(InlineBytecodeGenerator.java:280)
	at org.mockito.internal.creation.bytebuddy.InlineBytecodeGenerator.mockClass(InlineBytecodeGenerator.java:217)
	at org.mockito.internal.creation.bytebuddy.TypeCachingBytecodeGenerator.lambda$mockClass$0(TypeCachingBytecodeGenerator.java:47)
	at net.bytebuddy.TypeCache.findOrInsert(TypeCache.java:168)
```

**Comment by raphw** [comment]
That is strange. Could you run the issue with `-Dnet.bytebuddy.dump=/some/folder` so that I could have a look? If it really only happens on this version, it might just be a bug in the verifier of this version, too.

**Comment by aswath80** [comment]
sorry missed this update. Here is the output

`java.io.Serializable.1693369008524.class
java.io.Serializable-original.1693369008524.class
java.lang.ClassLoader$ByteBuddyAccessor$V1.1693369008152.class
java.lang.Comparable.1693369008522.class
java.lang.Comparable-original.1693369008522.class
java.lang.constant.Constable.1693369008526.class
java.lang.constant.Constable-original.1693369008526.class
java.lang.Enum.1693369008421.class
java.lang.Enum-original.1693369008421.class
net.bytebuddy.mirror.AccessibleObject.1693369007967.class
net.bytebuddy.mirror.AccessibleObject-original.1693369007967.class
net.bytebuddy.utility.Invoker$Dispatcher.class
org.mockito.internal.util.reflection.InstrumentationMemberAccessor$Dispatcher$ByteBuddy$OxUuqWH7.1693369008349.class
Test$TestEnum$1.1693369008528.class
Test$TestEnum$1-original.1693369008528.class
`

**Comment by aswath80** [comment]
[bytebuddy-debug.zip](https://github.com/raphw/byte-buddy/files/12471699/bytebuddy-debug.zip)

**Comment by raphw** [comment]
The only suspicious bit is a change of the enclosing method attribute which is missing. But even then it should not affect the verification.

Does this only happen in Azule and this particular version? I would assume this is a big in that VM.

**Comment by raphw** [comment]
Then this must be an issue in the verifier.

**Comment by aswath80** [comment]
Azul zing fixed the issue with the verifier.

```
Fixes for this issue have been included in version 23.02.401 release, which have been also included in our latest 23.02.500 stable release.  It is also schedule to be included in our 23.08.101 release as well. 
```
